src/context/AuthContext.tsx

```diff
@@ -65,24 +67,55 @@ export const AuthProvider: FC<{ children: ReactNode }> = ({ children }) => {
             .catch((error) => {
                 throw new Error('Error during logout:', error);
             })
+        // return await makeRequest('POST', `logout`, {})
+        //     .then(() => {
+        //         localStorage.removeItem('userId');
+        //         localStorage.removeItem('accessToken');
+        //         setUser(undefined);
+        //     })
+        //     .catch((error) => { throw new Error(error) });
     }
 
     const isAuth = async (userId: string) => {
-        await axios.post(`http://localhost:4000/login"/${userId}`)
-            .then((response) => {
-                setUser(response.data)
+        // await axios.post(`http://localhost:4000/login"/${userId}`)
+        //     .then((response) => {
+        //         setUser(response.data)
+
+        //     })
+        //     .catch((error) => {
+        //         throw new Error(error);
+        //     });
+        return await makeRequest('POST', `login"/${userId}`)
+            .then((res) => setUser(res?.data))
+            .catch((error) => { throw new Error(error) });
+    }
 
-            })
-            .catch((error) => {
-                throw new Error(error);
-            });
+    const fetchSingleClient = async (userId: string): Promise<EditUserType> => {
+        return await makeRequest('GET', `/getUserById/${userId}`)
+            .then((res) => res?.data)
+            .catch((err) => { throw new Error(err) });
+    }
 
+    const updateClient = async (user: EditUserType) => {
+        const { _id, ...rest } = user;
+        const token = localStorage.getItem('accessToken');
+        await axios.put(`http://localhost:4000/updateUser/${_id}`, rest, {
```

Comment:
czemu tu nie uzywasz hook'a od makeRequest?

Reply:
to jest zrobione na branchy 'fix-lazy-loading-and-requests' po marge'u do maina będzie użyte :)